DOCS-2262: Add imu-wit-hwt905 (#2911)
viamrobotics/docs: docs/components/movement-sensor/imu/imu-wit-hwt905.md

---
title: "Configure a WitMotion HWT905-TTL IMU"
linkTitle: "imu-wit-hwt905"
weight: 10
type: "docs"
description: "Configure a WitMotion HWT905-TTL IMU on your machine. Once configured, use the API to obtain the AngularVelocity, Orientation, CompassHeading and LinearAcceleration."
images: ["/icons/components/imu.svg"]

> # SMEs: Bucket Team

Reviewer: This is true, but specifically on the bucket team, Susmita knows the most. Worth mentioning here? I can go either way.

Author: ```suggestion
# SMEs: Susmita, Bucket Team
```

---

> An [inertial measurement unit (IMU)](https://en.wikipedia.org/wiki/Inertial_measurement_unit) provides data for the [`AngularVelocity`](/components/movement-sensor/#getangularvelocity), [`Orientation`](/components/movement-sensor/#getorientation), [`CompassHeading`](/components/movement-sensor/#getcompassheading), [`LinearAcceleration`](/components/movement-sensor/#getlinearacceleration), and [`GetAccuracy`](/components/movement-sensor/#getaccuracy) methods.

Reviewer: A possibly off-topic comment: I tried clicking on these from [1], and they all link to places half a page up from where the topic actually starts, so I accidentally start reading the topic just before the one I clicked on. It's as though clicking on these tries to get the relevant section to be about a third of the way from the bottom of my browser window, rather than at the top of the window. Is that a mistake, or on purpose?

[1] https://docs-test.viam.dev/2911/components/movement-sensor/imu/imu-wit-hwt905/

Author: Yeah…. @npentrel has looked into this before I believe

Reviewer: kk. As long as it's not a surprise. If there are changes to be made, they belong in a separate PR.

> Acceleration and magnetometer data are available by using the [sensor](/components/sensor/) [`GetReadings`](/components/sensor/#getreadings) method, which IMUs wrap.

Reviewer 2: Isn't it all movement sensors who wrap that?

Author: Well yes either way of wording it is still true. This is the way we word it on the other movement sensor pages--would you like me to change it to "which movement sensors wrap" on all of them?

Reviewer: The way this is phrased, it sounds like `GetReadings` will give you acceleration and magnetometer stuff but not give you angular velocity or orientation. I had assumed `GetReadings` would give you all the readings the hardware can provide, but I haven't actually tried it out myself to double-check that.

Author: Ooh yeah I was keeping consistent with the other page but @penguinland do you think if it said “accelerometer” instead of “acceleration” that’d work?

Reviewer: We should wire one of these up and try it out. I had expected it would give accelerometer, angular velocity, compass heading, and orientation. Now that I look at the code, I suspect it does _not_ give raw magnetometer data. but the real proof is in trying one out for real.

Reviewer: ![Screenshot_2024-05-14_17-14-27](https://github.com/viamrobotics/docs/assets/2861124/484e9dc4-f704-45e7-bc2e-0f9e5b699eaf)
![Screenshot_2024-05-14_17-14-11](https://github.com/viamrobotics/docs/assets/2861124/ce2f1b61-0604-43a0-8ec2-60fd421ff0d4)
Here are screenshots of what `GetReadings` and the main RC card look like. So, that's the data it can provide.

Author: Thanks so much for testing! So to me, it still seems like it'd be okay to say "accelerometer and magnetometer data" without getting overly specific. Or, do we like the following better?
```suggestion
To get orientation, angular velocity, linear acceleration and compass heading data, you can use the [sensor](/components/sensor/) [`GetReadings`](/components/sensor/#getreadings) method, which movement sensors inherit from the general sensor API.
```

Reviewer: Either one LGTM!

Navigate to the **CONFIGURE** tab of your machine's page in [the Viam app](https://app.viam.com).
Click the **+** icon next to your machine part in the left-hand menu and select **Component**.
Select the `movement-sensor` type, then select the `imu-wit-hwt905` model.
Enter a name or use the suggested name for your movement sensor and click **Create**.

{{< imgproc src="/components/movement-sensor/imu-wit-hwt905-builder.png" alt="Creation of an `imu-wit` movement sensor in the Viam app config builder." resize="1200x" style="width:650px" >}}

Then fill in the attributes as applicable to your movement sensor, according to the table below.

{{% /tab %}}
{{% tab name="JSON Template" %}}

```json {class="line-numbers linkable-line-numbers"}
{
  "components": [
    {
      "name": "<your-sensor-name>",
      "model": "imu-wit-hwt905",
      "type": "movement_sensor",
      "namespace": "rdk",
      "attributes": {
        "serial_path": "<your-port>",
        "serial_baud_rate": <int>
      },
      "depends_on": []
    }
  ]
}
```

{{% /tab %}}
{{% tab name="JSON Example" %}}

```json {class="line-numbers linkable-line-numbers"}
{
  "components": [
    {
      "name": "myIMU",

>       "model": "imu-wit",

Author: ```suggestion
      "model": "imu-wit-hwt905",
```
